docs/decisions/0001-modular-frontend.rst

```diff
@@ -0,0 +1,68 @@
+# Modular frontends
+
+## Background
+
+A couple of issues have been identified with the current MFE architecure that
+we hope to resolve with this proposal. Those issues are:
+
+- Duplicated javascript and styling code across multiple MFEs that needs to be
+  loaded multiple times.
+- Large bundle size for each MFE that includes the entire codebase instead of
+  the smaller subset of features needed.
+- Use of different libraries, versions of libraries etc across MFEs further
+  increasing overall bundle size.
+- An increasing number of different MFEs to keep track of, build and update.
+- Poor customisability due to the lack of plugins, or other hooks into MFEs
+- No seamless routing between MFEs.
+
+## Proposal
+
+Our proposal boils down to the following:
+
+- Creating of an LMS mono-repo that pulls in MFEs related to the LMS into a
+  single repo, potentially starting as submodules and eventully being hosted in
+  the main repo.
+- Use of webpack module federation to provide both a mechanism to support loose
+  dependencies between MFEs, and for plugins
+- Updating frontend-platform to support a single shared global state for auth
+  etc, and an mfe-level state.
+- Update to react-route v6 or alternative that supports relative routes, so that
+  the parent MFE can provide a base route, and the MFE can handle routing inside
+  that.
+- Use of npm workspaces to have each MFE continue to be a separate package but
+  dependencies shared across all MFEs can be added to the parent package and
+  updated globally there.
+- All MFEs canbe built in a single package, but can deployed separately.
```

Comment:
Typo: `canbe`. Do we actually need to support separate deployment? What about just being able to enable/disable them individually?

Reply:
I think here I means that each MFE will be a separate package, which can be built together or separately. Each can be deployed separately or together, and you can choose which to build and deploy.